src/main/java/org/gridsuite/directory/server/DirectoryController.java

```diff
@@ -76,6 +77,33 @@ public ResponseEntity<List<ElementAttributes>> getPath(@PathVariable("elementUui
         return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(service.getPath(elementUuid, userId));
     }
 
+    @PostMapping(value = "/elements/stash")
+    @Operation(summary = "Stash elements")
+        @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "elements were stashed")})
+    public ResponseEntity<Void> stashElements(@Parameter(description = "elements UUIDs") @RequestParam("ids") List<UUID> elementsUuid,
+                                              @RequestHeader("userId") String userId) {
+        service.stashElements(elementsUuid, userId);
+        return ResponseEntity.ok().build();
+    }
+
+    @PostMapping(value = "/elements/{parentUuid}/restore")
+    @Operation(summary = "Restore elements")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "elements were restored ")})
+    public ResponseEntity<Void> restoreElements(@RequestBody List<UUID> elementsUuid,
+                                                @PathVariable("parentUuid") UUID parentUuid,
+                                                @RequestHeader("userId") String userId) {
+        service.restoreElements(elementsUuid, parentUuid, userId);
+        return ResponseEntity.ok().build();
+    }
+
+    @GetMapping(value = "/elements/stash")
+    @Operation(summary = "Get the list of elements in the trash")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "the list of nodes in the trash")})
+    public ResponseEntity<List<Pair<ElementAttributes, Long>>> getStashedElements(@RequestHeader("userId") String userId) {
```

Comment:
Is it deliberate that the bin is global and not specific to each folder?

Reply:
Yes, it display all elements that are public or created by the user and when we restore elements, we restore them in the selected directory